File: palworld_info_server/src/pals/entities/pal.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, JoinTable, ManyToMany, ManyToOne, OneToMany } from "typeorm";
import { Element } from "./element.entity";
import { LevelWorkSuitability, WorkSuitability } from "./work_suitability.entity";
import { PSkillPal } from '../../skills/entities/passive_skill.entity';
import { PartnerPal } from "./partner.entity";
import { ActiveSkill } from "../../skills/entities/active_skill.entity";
import { ActiveSkillPal } from "../../skills/entities/active_skill_pal.entity";
import { PalItemEntity } from './pal_item.entity';

export enum PalSize {
  S = "S",
  M = "M",
  L = "L",
}

@Entity()
export class Pal {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({unique: true})
  name: string;

   @OneToMany(() => PalItemEntity, palItem => palItem.pal)
    palItems: PalItemEntity[];

  @Column({nullable: true})
  title: string;

  @Column({nullable: true})
  slug: string;

  @ManyToMany(() => Element)
  @JoinTable()
  elements: Element[]
  //
  @ManyToMany(() => LevelWorkSuitability)
  @JoinTable()
  levelWorkSuitability: LevelWorkSuitability[]

  @ManyToMany(() => PSkillPal, (pSkill) => pSkill.pals)
  @JoinTable()
  pSkillPals: PSkillPal[];

  @ManyToOne(() => PartnerPal, (partner) => partner.pal)
  partnerPal: PartnerPal;

  @ManyToMany(() => ActiveSkillPal, (activeSkillsPal) => activeSkillsPal.pals)
  @JoinTable()
  activeSkillsPal: ActiveSkillPal[];

  // @Column({
  //   type: "enum",
  //   enum: PalSize,
  //   default: PalSize.M
  // })
  // size: PalSize;

  @Column({nullable: true})
  iconUrl: string;

  @Column({nullable: true})
  summary: string;

  @Column({nullable: true})
  size: string;

  @Column({nullable: true})
  rarity: number;

  @Column({nullable: true})
  hp: number;

  @Column({nullable: true})
  meleeAttack: number;

  @Column({nullable: true})
  magicAttack: number;

  @Column({nullable: true})
  defense: number;

  @Column({nullable: true})
  support: number;

  @Column({nullable: true})
  craftSpeed: number;

  @Column({nullable: true})
  captureRate: number;

  @Column({nullable: true})
  price: number;

  @Column({nullable: true})
  slowWalkSpeed: number;

  @Column({nullable: true})
  runSpeed: number;

  @Column({nullable: true})
  rideSprintSpeed: number;

  @Column({nullable: true})
  foodAmount: number;

  @Column({nullable: true})
  maleProbability: number;
}

File: palworld_info_app/lib/domains/pal_entity.dart



import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:palworld_info_app/domains/element_entity.dart';
import 'package:palworld_info_app/domains/level_work_suitability_entity.dart';
import 'package:palworld_info_app/domains/partner_skill_entity.dart';

import '../utils/constants.dart';
import 'p_skill_pal_entity.dart';
part 'pal_entity.freezed.dart';
part 'pal_entity.g.dart';

@freezed
class PalEntity with _$PalEntity {
  const factory PalEntity({
    int? id,
    String? name,
    String? title,
    String? slug,
    String? iconUrl,
    String? summary,
    Size? size,
    int? rarity,
    int? hp,
    int? meleeAttack,
    int? magicAttack,
    int? defense,
    int? support,
    int? craftSpeed,
    int? captureRate,
    int? price,
    int? slowWalkSpeed,
    int? runSpeed,
    int? rideSprintSpeed,
    int? foodAmount,
    int? maleProbability,
    List<ElementEntity>? elements,
    List<LevelWorkSuitabilityEntity>? levelWorkSuitability,
    List<PSkillPalEntity>? pSkillPals,
    PartnerSkillEntity? partnerPal,
  }) = _PalEntity;
  factory PalEntity.fromJson(Map<String, dynamic> json) => _$PalEntityFromJson(json);
}

